pkg-manager/resolve-dependencies/src/resolveDependencyTree.ts

```diff
@@ -234,6 +238,28 @@ export async function resolveDependencyTree<T> (
   const { pkgAddressesByImporters, time } = await resolveRootDependencies(ctx, resolveArgs)
   const directDepsByImporterId = zipObj(importers.map(({ id }) => id), pkgAddressesByImporters)
 
+  let newDefaultCatalogs: Record<string, ResolvedCatalogEntry> | undefined
+  for (const directDependencies of pkgAddressesByImporters) {
+    for (const directDep of directDependencies as PkgAddress[]) {
+      const { alias, normalizedBareSpecifier, version, saveCatalog } = directDep
+      const existingCatalog = opts.catalogs?.default?.[alias]
+      if (existingCatalog != null) {
+        if (existingCatalog !== normalizedBareSpecifier) {
+          globalWarn(
+            `Skip adding ${alias} to the default catalog because it already exists as ${existingCatalog}, please use \`pnpm update\` to update the catalogs.`
```

Comment:
The `pnpm update` command still doesn't support catalogs. Sorry, that's my fault and it's still on my plate. I want to finish that soon this weekend.

Reply:
I have remove the part about `pnpm update` from the message. You may add them on your own PR when you implement update with catalog.